### PR TITLE
Reenabled android tests and moved ExtendedAsynchronousByteChannel

### PR DESCRIPTION
**Pull request by rozza**
Moved ExtendedAsynchronousByteChannel to com.mongodb.internal.connection
The dex'ing process didn't like it being referenced in a higher package
when the code was split across dex files.

JAVA-3142

### CODE REVIEW

**Comment by rozza** [comment]
*That* was fun!

Patch build: https://evergreen.mongodb.com/version/5c41e52aa4cf47124dc21b59

File: driver-embedded-android/build.gradle [hunk 1]
```diff
@@ -64,6 +64,9 @@ android {
         release {
             minifyEnabled false
         }
+        debug {
```
Comment (jyemin) [inline driver-embedded-android/build.gradle]: Just confirming we want to keep this and it wasn't meant to be temporary while you debugged.

File: driver-core/src/main/com/mongodb/internal/connection/ExtendedAsynchronousByteChannel.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
  * https://github.com/marianobarrios/tls-channel
  */
 
-package com.mongodb.internal.connection.tlschannel.async;
+package com.mongodb.internal.connection;
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/connection/ExtendedAsynchronousByteChannel.java]: Funny, I thought about moving this up when AsynchronousSocketChannelStream started to use it, but then didn't do it.

File: driver-embedded-android/build.gradle (hunk 1 above)
Comment (rozza) [reply driver-embedded-android/build.gradle]: Ah that can now be removed.

**Pull request closed without merging**